### PR TITLE
feature/ATC-77

### PR DESCRIPTION
**Pull request by n8rzz**
completes #77 

Adds initial code for ModelSourceFactory and ModelSourcePool

### CODE REVIEW

**Comment by n8rzz** [comment]
@erikquinn I'm about half way through this one but I think there is enough here for you tho get an idea of where this is headed.  This one might blow your hair back a little bit 😉 , so ask questions on the stuff that doesn't make sense.

**Comment by n8rzz** [comment]
@erikquinn this is as good as it's going to get before I finish #53.  I added issues #120  and #121 for future work that can't be completed now, mostly because of #53.

Take a look and give this a test drive, I want to make sure I didn't break anything.  My tests show that I haven't, but I like to be sure.

File: src/assets/scripts/base/ModelSource/ModelSourcePool.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+        this._items.push(modelToAdd);
+    }
+
+    // TODO: this method currenty accepts only a string. It may be easier to pass the constructor
+    /**
+     * Remove a reusable from the collection and return it for use by the caller
+     *
+     * @for ModelSourcePool
+     * @method releasModelFromPool
```
Comment (erikquinn) [inline src/assets/scripts/base/ModelSource/ModelSourcePool.js]: `releaseReusable`

File: src/assets/scripts/base/ModelSource/ModelSourcePool.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+     *
+     * When a reusable has been requested, it mush be removed from the pool
+     *
+     * @for ModelSourcePool
+     * @method _removeItem
+     * @param model {constructor} one of CLASS_MAP
+     * @private
+     */
+    _removeItem(model) {
+        this._items = _without(this._items, model);
```
Comment (erikquinn) [inline src/assets/scripts/base/ModelSource/ModelSourcePool.js:150]: :+1: that's a neat lodash function... it will come in handy :smile:

**Review by erikquinn — APPROVED** [review]
I'm not sure if it's related to this specifically or not, but the only weirdness I noticed was that upon rerouting an aircraft (in this case from `BSR.BSR2.KSFO` to `QUINN.BDEGA2.KSFO`, they will climb of their own accord to very high altitude and not come down. They can be told to descend normally, but a strange altitude is being put in there. May or may not be related.

That aside, everything relating to fixes seems to be working properly, which, as I saw it in review, is the only place the reusability factory is put to work.

File: src/assets/scripts/base/ModelSource/ModelSourcePool.js (hunk 2 above)
Comment (n8rzz) [reply src/assets/scripts/base/ModelSource/ModelSourcePool.js:150]: And it doesn't mutate `_items` it returns an entirely new array.

File: src/assets/scripts/base/ModelSource/ModelSourcePool.js (hunk 1 above)
Comment (n8rzz) [reply src/assets/scripts/base/ModelSource/ModelSourcePool.js]: done

I'm so glad I have you look over these PRs for me.

**Comment by n8rzz** [comment]
@erikquinn added #155 for the bug you reported.  It may be something, it may not, but at least we have visibility.

**Pull request merged**